bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLine.java

```diff
@@ -358,10 +358,12 @@ public void beginTask(String name, int totalWork) {
 		// make sure the progress bar is made visible while
 		// the task is running. Fixes bug 32198 for the non-animated case.
 		Runnable timer = () -> StatusLine.this.startTask(timestamp, animated);
-		inUIThread(() -> {
-			fProgressBar.getDisplay().timerExec(DELAY_PROGRESS, timer);
-			if (!animated) {
-				fProgressBar.beginTask(totalWork);
+		getDisplay().execute(() -> {
+			if (fProgressBar != null && !fProgressBar.isDisposed()) {
```

Comment:
another example where it would be more handsome to have a fProgressBar.execute()

Reply:
sure, but can we please focus on the issue here? It actually would be even more usefull if we would not need the disposal check at all as it is mostly useless anyways instead of wrapping more code around this, all of this is not related to the change present here...